### PR TITLE
Use macholib and pefile from PyPI

### PR DESCRIPTION
**Pull request by xoviat**

### CODE REVIEW

File: tests/requirements-tools.txt [hunk 1]
```diff
@@ -12,4 +12,4 @@ git+https://github.com/xoviat/pytest-plugins.git@common_setup#egg=pytest-profili
 
 psutil  # Better subprocess alternative with implemented timeout.
 
-https://github.com/erocarrera/pefile/archive/develop.zip  # contains changes not yet released
+git+https://github.com/xoviat/pefile.git@patch-1  # contains optimizations to be merged
```
Comment (htgoebel) [inline tests/requirements-tools.txt]: Optimization are merges meanwhile, see https://github.com/pyinstaller/pyinstaller/issues/1920#issuecomment-310918505. If we remove the bundled pefile, please change back to `…/erocarrera/pefile/archive/develop.zip`

**Comment by htgoebel** [comment]
I'm not sure if I understand what this Pull-request is about. The commit-messages are quite terse. It looks like this is updating macholib but also switching to the pefile package from your repo without actually removing the bundled pefile. But contrary the requirement `pefile` is not restored in `setup.py`.

Can you please explain? (And please rebase on current development head to get rid of the unrelated changes.)

**Comment by xoviat** [comment]
> Can you please explain? (And please rebase on current development head to get rid of the unrelated changes.)

This is currently pending on erocarrera/pefile#183 and a release. The goal is to go back to the pypi packages.

**Comment by xoviat** [comment]
I am sorry but that PR is over a month old, so I thought it would have been merged by now. @erocarrera seems to be busy.

**Comment by htgoebel** [comment]
> I am sorry but that PR is over a month old, so I thought it would have been merged by now. 

Maybe. But what is *this* pull-request about? See https://github.com/pyinstaller/pyinstaller/pull/2589#issuecomment-318776519

**Comment by xoviat** [comment]
> Maybe. But what is this pull-request about? See #2589 (comment)

This is about moving back onto pypi macho and pefile. This cannot be done until my PR on pefile is merged.

**Comment by xoviat** [comment]
OK this is cleaned up. However, merging this introduces a dependency on an unreleased pypi pefile that I was assured would be released soon.

**Comment by xoviat** [comment]
OK to merge as of one hour ago; pefile has been released.

**Comment by htgoebel** [comment]
Thanks, not it makes more sense :-)

What is the reason for installing requirements first d0c9a9d5516e72627e0abcd?

**Comment by xoviat** [comment]
> What is the reason for installing requirements first d0c9a9d?

There is currently no particular reason but previously it was that pefile was unreleased.

File: .gitignore [hunk 2]
```diff
@@ -87,3 +87,4 @@ tests/old_suite/basic/test_pyz_as_external_file.exe.pkg
 
 # py.test cache folder
 .cache
+*.prof
```
Comment (htgoebel) [inline .gitignore]: Missing new line.

File: PyInstaller/depend/bindepend.py [hunk 3]
```diff
@@ -15,23 +14,25 @@
...
 from . import dylib, utils
 
-
 from .. import log as logging
 from ..utils.win32 import winutils
 
 logger = logging.getLogger(__name__)
 
 seen = set()
+pefile.fast_load = True
```
Comment (htgoebel) [inline PyInstaller/depend/bindepend.py]: Please add a comment why we need this and what the effect is.

File: setup.py [hunk 4]
```diff
@@ -16,8 +16,11 @@
 from PyInstaller import (__version__ as version, is_linux, is_win, is_cygwin,
                          HOMEPATH, PLATFORM, compat)
 
-
-REQUIREMENTS = ['setuptools']
+REQUIREMENTS = [
+    'setuptools',
+    'pefile > 2016.3.28',
```
Comment (htgoebel) [inline setup.py]: Shouldn't we require >= 2017.8.1?

Comment (xoviat) [inline setup.py]: This is already configured correctly because 2016.3.28 does not satisfy this.

File: .gitignore (hunk 2 above)
Comment (xoviat) [inline .gitignore]: Done.

**Comment by htgoebel** [comment]
@xoviat I merged this manually after some cleanup. Thanks for your work!

**Pull request closed without merging**